ui/src/components/chip/__tests__/QChip.spec.js

```diff
@@ -1,154 +1,467 @@
+        })
+
+        it('should NOT be selected when selected is false', () => {
+          mountQChip({
+            props: {
+              selected: false
+            }
+          })
+
+          // q-ship should not have the class "q-chip--selected"
+          cy.get('.q-chip').not('.q-chip--selected').should('exist')
         })
       })
     })
 
     describe('Category: state', () => {
       describe('(prop): clickable', () => {
-        it.skip(' ', () => {
-          //
+        it('should have hover effects and emit click events when clickable is true', () => {
+          const fn = cy.stub()
+
+          mountQChip({
+            props: {
+              clickable: true,
+              onClick: fn
+            }
+          })
+
+          cy.get('.q-chip')
+            .should('have.css', 'cursor', 'pointer')
+            .click()
+            .then(() => expect(fn).to.be.calledOnce)
         })
       })
 
       describe('(prop): removable', () => {
-        it.skip(' ', () => {
-          //
+        it('should displays a "remove" icon that when clicked the QChip emits "remove" event', () => {
+          const fn = cy.stub()
+
+          mountQChip({
+            props: {
+              removable: true,
+              onRemove: fn
+            }
+          })
+
+          cy.get('.q-chip')
+            .get('.q-icon.q-chip__icon--remove')
+            .click()
+            .then(() => expect(fn).to.be.calledOnce)
         })
       })
 
       describe('(prop): disable', () => {
-        it.skip(' ', () => {
-          //
+        it('should NOT have hover effects and NOT emit click events when disable is true', () => {
+          const fn = cy.stub()
+
+          mountQChip({
+            props: {
+              disable: true,
+              onClick: fn
+            }
+          })
+
+          cy.get('.q-chip')
+            .should('not.have.css', 'cursor', 'pointer')
+            .click()
+            .then(() => expect(fn).to.not.be.called)
         })
       })
     })
 
     describe('Category: style', () => {
       describe('(prop): dense', () => {
-        it.skip(' ', () => {
-          //
+        it('should have a dense style when dense is true', () => {
+          mountQChip({
+            props: {
+              dense: true
+            }
+          })
+
+          cy.get('.q-chip.q-chip--dense').should('exist')
         })
       })
 
       describe('(prop): size', () => {
-        it.skip(' ', () => {
-          //
+        it('should change QChip size based on a CSS unit value', () => {
+          const size = '50px'
+
+          mountQChip({
+            props: {
+              size
+            }
+          })
+
+          cy.get('.q-chip').should('have.css', 'font-size', size)
+        })
+
+        it('should change QChip size based defined values: xs, sm, md, lg, xl', () => {
+          const sizes = [ 'xs', 'sm', 'md', 'lg', 'xl' ]
+
+          sizes.forEach((size) => {
```

Comment:
Instead of mounting a new one each time, mount it once, then use the component global reference to change the prop, so you can avoid removing the component for each cycle

Also, rely on plain `for` when possible (`for of` in this case)